### PR TITLE
Allow Choices Option

### PR DESCRIPTION
**Pull request by devotox**
This allows us to have an array of possible choices and if the value is not one of those choices in the array an error is thrown

### CODE REVIEW

**Review by bill-edwards — APPROVED** [review]
Looks good to me.

File: lib/model.js [hunk 1]
```diff
@@ -1,9 +1,80 @@
...
+	console.log(TestModel.isValid());
+
+	console.log('\nValidate'); // Throws Error
+	console.log(TestModel.validate());
+*/
+
 const ModelValidator = require('./model_validator');
 const ModelUtilities = require('./model_utilities');
 
 const flattenArray = ModelUtilities.flattenArray;
```
Comment (thehogfather) [inline lib/model.js]: for your next changes, destructuring is your friend :)

File: lib/model.js [hunk 2]
```diff
@@ -121,14 +192,14 @@ module.exports = class Model {
 			let type = attribute && exists(attribute.type) && attribute.type;
 			let transform = attribute && isFunction(attribute.transform) && attribute.transform;
 
-			let value = values && exists(values[key]) ? values[key] : null;
+			let value = values && exists(values[key]) ? values[key] : undefined;
 			value = ( type !== 'function' && isFunction(value) ) ? value(key, type) : value;
```
Comment (thehogfather) [inline lib/model.js:197]: just checking this is the intended logic - type is not function so we go ahead and invoke it as a function?

**Review by thehogfather — COMMENTED** [review]
one question about the intended logic  when `type is not function`. Otherwise looks fine.

File: lib/model.js [hunk 3]
```diff
@@ -229,81 +300,13 @@ module.exports = class Model {
 		if( attr.required && !exists(this[key])) { return `${key} of type ${attr.type} is required`; }
 
 		// Validate type and value is of type
-		if(ModelValidator[attr.type](this[key], attr.options)) { if(!attr.custom) { return; } }
-		else { return `${key} should be of type ${attr.type}`; }
+		if(!ModelValidator[attr.type](this[key], attr.options)){ return `${key} should be of type ${attr.type}`; }
```
Comment (rafaelalvessa) [inline lib/model.js]: Add a space before `{`.

File: lib/model.js [hunk 4]
```diff
@@ -229,81 +300,13 @@ module.exports = class Model {
...
+
+		if( attr.choices && attr.choices.length &&
```
Comment (rafaelalvessa) [inline lib/model.js]: Remove extra space before `attr.choices`.

File: lib/model.js [hunk 5]
```diff
@@ -229,81 +300,13 @@ module.exports = class Model {
...
+			isArray(attr.choices) && !attr.choices.includes(this[key]))
+			{ return `${key} of type ${attr.type} should be one of ${attr.choices}`; }
```
Comment (rafaelalvessa) [inline lib/model.js]: Fix indentation in lines 306-307.

**Review by rafaelalvessa — CHANGES_REQUESTED** [review]

File: lib/model.js (hunk 2 above)
Comment (devotox) [reply lib/model.js:197]: yes basically if the type is function then it means we should test the value to be a function but if its not a function and it sees value as a function it runs it

**Pull request merged**